pkg/plugins/policies/meshpassthrough/plugin/xds/listeners_filter_chain.go

```diff
@@ -155,8 +155,10 @@ func (c FilterChainConfigurer) configureTcpFilterChain(listener *envoy_listener.
 		WithClusterName(clusterName).
 		Build()
 	filterChainBuilder := xds_listeners.NewFilterChainBuilder(c.APIVersion, chainName).
-		Configure(xds_listeners.TCPProxy(clusterName, split)).
-		Configure(xds_listeners.MatchTransportProtocol("raw_buffer"))
+		Configure(xds_listeners.TCPProxy(clusterName, split))
+	if c.Protocol != core_mesh.ProtocolMysql {
+		filterChainBuilder.Configure(xds_listeners.MatchTransportProtocol("raw_buffer"))
+	}
```

Comment:
> raw_buffer - default, used when no transport protocol is detected,

Will the conditional make no sense?

https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/listener/v3/listener_components.proto